packages/core/src/utils/saml-assertion-handler.ts

```diff
@@ -60,3 +65,57 @@ export const getConnectorSessionResultFromJti = async (
 
   return connectorSessionResult.data.connectorSession;
 };
+
+/**
+ * Get the single sign on session data from the oidc provider session storage by the jti.
+ *
+ * @param jti The jti of the interaction session.
+ * @param provider
+ * @returns SingleSignOnConnectorSession
+ *
+ * @remark This method is used by the SSO SAML assertion consumer service endpoint.
+ * Since we do not have the interaction ctx under SAML ACS endpoint, we need to get the session data by the jti.
+ *
+ * Forked from the above @see `getConnectorSessionResultFromJti` method, with more detailed SingleSignOnConnectorSession type guard.
+ * The above method will be deprecated in the future.
```

Comment:
we can directly mark the above method `@deprecated`?

Reply:
Can't for now, as it is still being used by the SAML standard social connector.

Reply:
deprecated is not removed.

Reply:
i mean even though a method is marked as deprecated it can be used for a while

Reply:
cool

Reply:
tag added